### PR TITLE
Memory leak when input is not a numpy array

### PR DESCRIPTION
**Pull request by kwgoodman**
All bottleneck functions leak memory, as reported in #201, when the input is not a numpy array.

### CODE REVIEW

File: bottleneck/src/reduce_template.c [hunk 1]
```diff
@@ -1253,6 +1255,7 @@ reducer(char *name,
         axis = PyArray_PyIntAsInt(axis_obj);
         if (error_converting(axis)) {
             TYPE_ERR("`axis` must be an integer or None");
+            Py_DECREF(a);
```
[the remark this answers is not in the document]
Comment (kwgoodman) [reply bottleneck/src/reduce_template.c]: I don't find the goto more readable but going along with what numpy does is compelling

**Comment by kwgoodman** [comment]
I removed static and added the goto. I wasn't ready to change ``y`` to ``result`` because lower level functions also use ``y``.

File: bottleneck/src/reduce_template.c [hunk 2]
```diff
@@ -1290,41 +1292,49 @@ reducer(char *name,
...
+            y = slow(name, args, kwds);
         }
 
     }
 
+    Py_DECREF(a);
+
+    return y;
+
+    cleanup:
```
Comment (shoyer) [inline bottleneck/src/reduce_template.c]: nit: this is typically dedented (at least in the NumPy code I've seen). Also if you're only using it for error handling, I might call it `error` instead.

**Comment by shoyer** [comment]
Looks great!

**Pull request merged**